# 2BP: Secondary Bacteria Pneumonia in COVID-19

This repository contains the code for the paper **Microbial Dynamics and Pulmonary Immune Responses in COVID-19 Secondary Bacterial Pneumonia**.

## Code

2BPCode_final.R used to analyze data at the microbe level and produce manuscript figures 2, 3, and 4. This code uses the following input files: 
* [TA_samples_postbacterialQC_longitudinal.csv](Inputs_Final/TA_samples_postbacterialQC_longitudinal.csv): a list of tracheal aspirate samples for all patients at all timepoints available, after bacterial QC.
* [TA_samples_postbacterialANDhostQC_timepoint.csv](Inputs_Final/TA_samples_postbacterialANDhostQC_timepoint.csv): a list of tracheal aspirate samples at timepoint closest to 2BP diagnosis, or, for noBP controls, at equivalent time on ventilator. After host and bacterial QC. 
* [NS_samples_postbacterialQC_longitudinal.csv](Inputs_Final/NS_samples_postbacterialQC_longitudinal.csv): a list of nasal swab samples for all patients at all timepoints available, after bacterial QC.
* [NS_samples_postbacterialQC_timepoint.csv](Inputs_Final/NS_samples_postbacterialQC_timepoint.csv): a list of nasal swab samples at timepoint closest to 2BP diagnosis, or, for noBP controls, at equivalent time on ventilator, after bacterial QC. 
* [Patient_data_noPHI.csv](Inputs_Final/Patient_data_noPHI.csv): patient metadata, deidentified.
* [2BP_CultureResults.csv](Inputs_Final/2BP_CultureResults.csv): culture results from conventional microbiologic assays for patients with 2BP.
* [Microbe_reports_genus.csv](Inputs_Final/Microbe_reports_genus.csv): outputs from CZID pipeline.
* [BetaCoV_nt_rpM.csv](Inputs_Final/BetaCoV_nt_rpM.csv): SARS-CoV-2 reads per million.
* [Analyzed_samples.csv](Inputs_Final/Analyzed_samples.csv): bacterial mass file.

2BP_AMRCode.Rmd is used to make manuscript figure 2H (beta diversity plot), and Figure 5 and Figure S3 (AMR resistance plots). This code uses the following additional input files: 
* [AMR_genes_of_interest.csv](Inputs_Final/AMR_genes_of_interest.csv): List of AMR genes of interest 
* [ARGANNOT_genelist.csv](Inputs_Final/ARGANNOT_genelist.csv): List of AMR genes from ARGANNOT databse
* [2BP_CultureResults_AMR.csv](Inputs_Final/2BP_CultureResults_AMR.csv): Culture results formatted for AMR analyses
* [idseq_amr_heatmap_values_final.csv](Inputs_Final/idseq_amr_heatmap_values_final.csv) : output of CZID AMR pipeline for this dataset


2BP_host_gene_analysis. R is used to analyze data at the human gene expression data and make figure 6 and Supp Tables S3-S7. This notebook uses the following additional input files: 
* [TA_samples_postbacterialANDhostQC_timepoint.csv](Inputs_Final/TA_samples_postbacterialANDhostQC_timepoint.csv): metadata file.
* [BetaCoV_nt_rpM.csv](Inputs_Final/BetaCoV_nt_rpM.csv): SARS-CoV-2 viral abundance, in reads per million.
* [host_gene_name_mapping.csv](Inputs_Final/host_gene_name_mapping.csv): for conversion between gene symbol and gene ensembl ID.
* [host_gene_counts.csv](Input_Finals/host_gene_counts.csv): host gene count table.
* [SupplementaryTable1B_InputData_011024.csv](Inputs_Final/SupplementaryTable1B_InputData_011024.csv): steroid information.
* [TA_samples_timepoint_bacterial_mass.csv](Input_Finals/TA_samples_timepoint_bacterial_mass.csv): total bacterial mass.

## Figures

### Main figures

Figure 2: csv output files for panels A, B, C, E, G were generated by 2BPCode_final.R, and then final graphs created in Prism. Figure 2D and 2H were created as graphs in R and final edits performed in Prism (standardizing font size).  
* A: [TA_samples_timepoint_bacterial_mass.csv](Outputs_Final/TA_samples_timepoint_bacterial_mass.csv)
* B: [TA_samples_alpha_timepoint.csv](Outputs_Final/TA_samples_alpha_timepoint.csv)
* C: [TA_alpha_vs_rank_of_top_pathogen_at_toprankedsample.csv](Outputs_Final/TA_alpha_vs_rank_of_top_pathogen_at_toprankedsample.csv)
* D: [Figure2D.pdf](Outputs_Final/Figure2D.pdf)
* E: [exemplary_1154.csv](Outputs_Final/exemplary_1154.csv) and [exemplary_1204.csv](Outputs_Final/exemplary_1204.csv)
* G: [toppathogenrankedat_TA_timepoint_sample.csv](Outputs_Final/toppathogenrankedat_TA_timepoint_sample.csv)
* H: [beta_diversity.pdf](Outputs_Final/beta_diversity.pdf)

Figure 3: Graph generated by 2BPCode_final.R and then final edits performed in Illustrator (adding pathogen names, adding empty circles for the samples in which pathogen was not detected, adding antibiotic data)
* A: [Figure3.pdf](Outputs_Final/Figure3.pdf)

Figure 4: csv output files for panels A, B, C were generated by 2BPCode_final.R, and then final graphs created in Prism. Figure 4D was created as a graph in R and final edits performed in Illustrator (adding pathogen names, adding empty circles for samples in which pathogen was not detected). 
* A: [NS_samples_mass_timepoint.csv](Outputs_Final/NS_samples_mass_timepoint.csv)
* B: [NS_samples_alpha_timepoint.csv](Outputs_Final/NS_samples_alpha_timepoint.csv)
* C: [NS_toprankofculturedpathogen.csv](Outputs_Final/NS_toprankofculturedpathogen.csv)
* D: [Figure 4D.pdf](Outputs_Final/Figure4D.pdf)

Figure 5: Created as a graph in R using code 2BP_AMRCode.Rmd and finalized in Illustrator (standardizing font sizes) 
* [heatmap_amr_all_labels.pdf](Outputs_Final/heatmap_amr_all_labels.pdf)

Figure 6: generated by the code 2BP_host_gene_analysis.R. The panels were exported under the following names:
* A: [bulk_vap_volcano_covariate-viralload.png](Outputs_Final/bulk_vap_volcano_covariate-viralload.png)
* B: [bulk_vap-vs-novap_gsea-hallmark.svg](Outputs_Final/bulk_vap-vs-novap_gsea-hallmark.svg)
* C: [bulk_vap-vs-novap_steroid_gsea-hallmark.svg](Outputs_Final/bulk_vap-vs-novap_steroid_gsea-hallmark.svg)
* D: [bulk_vap_mass_volcano.png](Outputs_Final/bulk_vap_mass_volcano.png)
* E: [bulk_vap_mass_genes.svg](Outputs_Final/bulk_vap_mass_genes.svg)
* F: [bulk_vap_mass_gsea-hallmark.svg](Outputs_Final/bulk_vap_mass_gsea-hallmark.svg)

### Supplementary figures

Supp Figure 1: generatedy by the code 2BPCode_final.R. csv files was generated by 2BPCode_final.R and then final graph created in Prism.
* [TA_samples_timepoint_sarscov2_ntrpm.csv](Final_Outputs/TA_samples_timepoint_sarscov2_ntrpm.csv)

Supp Figure 2: raw data provided in [Supplementary_Fig2_TA_persistor_notpersistor.xlsx](Final_Outputs/Supplementary_Fig2_TA_persistor_notpersistor.xlsx) and graphed in Prism.

Supp Figure 3:  Created as graphs in R using code 2BP_AMRCode.Rmd and finalized in Illustrator (standardizing font sizes)
* A: [AMR_plot_GNR.pdf](Outputs_Final/AMR_plot_GNR.pdf)
* B: [AMR_plot_GPC.pdf](Outputs_Final/AMR_plot_GPC.pdf)
* C: [AMR_plot_both.pdf](Outputs_Final/AMR_plot_both.pdf)

## Tables

Table S1: Clinical data, gathered from clinical record and deidentified.

Table S2: Antimicrobial susceptibilities from standard microbiological test, gathered from clinical record and deidentified.

Table S3: generated by the code 2BP_host_gene_analysis.R, saved as DE_host_vs_nohost_covariate-viralload.csv.

Table S4: generated by the code 2BP_host_gene_analysis.R, saved as gsea-hallmark_host_vs_nohost_covariate-viralload.tsv.

Table S5: generated by the code 2BP_host_gene_analysis.R, saved as DE_VAP_vs_noVAP_steroid_covariate-viralload.csv.

Table S6: generated by the code 2BP_host_gene_analysis.R, saved as gsea-hallmark_VAP_vs_noVAP_steroid_covariate-viralload.tsv.

Table S7: generated by the code 2BP_host_gene_analysis.R, saved as DE_host_mass.csv.

Table S8: generated by the code 2BP_host_gene_analysis.R, saved as gsea-hallmark_host_mass.tsv.

## Required hardware and software

The codes were run on a Mac laptop. The required R packages (see below for more details) could be installed with the command `install.package()` and `BiocManager::install()`. Each package can take up to 1 minute to install. Please refer to each package's website for more information on the installation.

### 2BPCode_final.R

```
R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] vegan_2.6-4     lattice_0.22-5  permute_0.9-7   lubridate_1.9.3 forcats_1.0.0   stringr_1.5.1  
 [7] purrr_1.0.2     readr_2.1.5     tidyr_1.3.0     tibble_3.2.1    ggplot2_3.4.4   tidyverse_2.0.0
[13] dplyr_1.1.4     limma_3.56.2   

loaded via a namespace (and not attached):
 [1] utf8_1.2.4        generics_0.1.3    stringi_1.8.3     lme4_1.1-35.1     hms_1.1.3        
 [6] digest_0.6.34     magrittr_2.0.3    timechange_0.3.0  evaluate_0.23     grid_4.3.0       
[11] fastmap_1.1.1     Matrix_1.6-5      mgcv_1.9-1        fansi_1.0.6       scales_1.3.0     
[16] cli_3.6.2         rlang_1.1.3       munsell_0.5.0     splines_4.3.0     withr_3.0.0      
[21] yaml_2.3.8        parallel_4.3.0    tools_4.3.0       tzdb_0.4.0        nloptr_2.0.3     
[26] minqa_1.2.6       colorspace_2.1-0  boot_1.3-28.1     vctrs_0.6.5       R6_2.5.1         
[31] lifecycle_1.0.4   MASS_7.3-60.0.1   cluster_2.1.6     pkgconfig_2.0.3   pillar_1.9.0     
[36] gtable_0.3.4      glue_1.7.0        Rcpp_1.0.12       xfun_0.41         tidyselect_1.2.0 
[41] rstudioapi_0.15.0 knitr_1.45        farver_2.1.1      htmltools_0.5.7   nlme_3.1-164     
[46] labeling_0.4.3    rmarkdown_2.25    compiler_4.3.0   
```

### 2BP_AMRCode.Rmd

```
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Ventura 13.2

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_1.0.0               stringr_1.4.1               dplyr_1.1.3                 purrr_1.0.2                 readr_2.1.3                
 [6] tidyr_1.2.1                 tibble_3.2.1                ggplot2_3.4.0               tidyverse_1.3.2             DESeq2_1.36.0              
[11] SummarizedExperiment_1.26.1 Biobase_2.56.0              MatrixGenerics_1.8.1        matrixStats_0.63.0          GenomicRanges_1.48.0       
[16] GenomeInfoDb_1.32.4         IRanges_2.30.1              S4Vectors_0.34.0            BiocGenerics_0.42.0         vegan_2.6-4                
[21] lattice_0.20-45             permute_0.9-7              

loaded via a namespace (and not attached):
 [1] googledrive_2.0.0      colorspace_2.0-3       ggsignif_0.6.4         ellipsis_0.3.2         XVector_0.36.0         fs_1.6.3              
 [7] rstudioapi_0.14        ggpubr_0.5.0           bit64_4.0.5            AnnotationDbi_1.58.0   fansi_1.0.3            lubridate_1.9.0       
[13] xml2_1.3.3             codetools_0.2-18       splines_4.2.1          cachem_1.0.6           geneplotter_1.74.0     knitr_1.41            
[19] jsonlite_1.8.3         broom_1.0.5            annotate_1.74.0        cluster_2.1.4          dbplyr_2.2.1           png_0.1-7             
[25] compiler_4.2.1         httr_1.4.4             backports_1.4.1        assertthat_0.2.1       Matrix_1.5-3           fastmap_1.1.0         
[31] gargle_1.2.1           cli_3.6.1              htmltools_0.5.6.1      tools_4.2.1            gtable_0.3.4           glue_1.6.2            
[37] GenomeInfoDbData_1.2.8 Rcpp_1.0.9             carData_3.0-5          cellranger_1.1.0       vctrs_0.6.4            Biostrings_2.64.1     
[43] nlme_3.1-160           xfun_0.40              rvest_1.0.3            timechange_0.1.1       lifecycle_1.0.3        rstatix_0.7.1         
[49] XML_3.99-0.12          googlesheets4_1.0.1    zlibbioc_1.42.0        MASS_7.3-58.1          scales_1.2.1           hms_1.1.2             
[55] parallel_4.2.1         RColorBrewer_1.1-3     yaml_2.3.6             memoise_2.0.1          stringi_1.7.8          RSQLite_2.2.19        
[61] genefilter_1.78.0      BiocParallel_1.30.4    rlang_1.1.1            pkgconfig_2.0.3        bitops_1.0-7           evaluate_0.18         
[67] bit_4.0.5              tidyselect_1.2.0       magrittr_2.0.3         R6_2.5.1               generics_0.1.3         DelayedArray_0.22.0   
[73] DBI_1.1.3              pillar_1.9.0           haven_2.5.1            withr_2.5.0            mgcv_1.8-41            survival_3.4-0        
[79] KEGGREST_1.36.3        abind_1.4-5            RCurl_1.98-1.9         modelr_0.1.10          crayon_1.5.2           car_3.1-1             
[85] utf8_1.2.2             tzdb_0.3.0             rmarkdown_2.18         locfit_1.5-9.6         grid_4.2.1             readxl_1.4.1          
[91] blob_1.2.3             reprex_2.0.2           digest_0.6.30          xtable_1.8-4           munsell_0.5.0     
```

### 2BP_host_gene_analysis.R

```
R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.4

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: US/Pacific
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] fgsea_1.28.0    limma_3.58.1    patchwork_1.2.0 lubridate_1.9.3 forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4    
 [8] purrr_1.0.2     readr_2.1.5     tidyr_1.3.1     tibble_3.2.1    ggplot2_3.4.4   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] Matrix_1.6-5        gtable_0.3.4        babelgene_22.9      compiler_4.3.2      Rcpp_1.0.12         tidyselect_1.2.0   
 [7] parallel_4.3.2      scales_1.3.0        BiocParallel_1.36.0 statmod_1.5.0       lattice_0.22-5      R6_2.5.1           
[13] generics_0.1.3      munsell_0.5.0       pillar_1.9.0        tzdb_0.4.0          rlang_1.1.3         fastmatch_1.1-4    
[19] utf8_1.2.4          stringi_1.8.3       msigdbr_7.5.1       timechange_0.3.0    cli_3.6.2           withr_3.0.0        
[25] magrittr_2.0.3      grid_4.3.2          rstudioapi_0.15.0   hms_1.1.3           cowplot_1.1.3       beeswarm_0.4.0     
[31] lifecycle_1.0.4     vctrs_0.6.5         glue_1.7.0          data.table_1.14.10  codetools_0.2-19    fansi_1.0.6        
[37] colorspace_2.1-0    tools_4.3.2         pkgconfig_2.0.3    
```

## Code usage

The full data sets are available in the folder [Inputs_Final](Inputs_Final). To reproduce the results in the paper, the uploaded codes could be run directly. The codes' output are located in the folder [Output_Finals](Output_Finals).
